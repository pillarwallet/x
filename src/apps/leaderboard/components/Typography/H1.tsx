import { ReactNode } from 'react';

type H1Props = {
  children: ReactNode;
  className?: string;
};

const H1 = ({ children, className }: H1Props) => {
  return (
    <h1
      className={`desktop:text-3xl desktop:leading-[45px] tablet:text-3xl tablet:leading-[45px] mobile:text-xl mobile:leading-5 font-medium ${className}`}
    >
      {children}
    </h1>
  );
};

export default H1;
