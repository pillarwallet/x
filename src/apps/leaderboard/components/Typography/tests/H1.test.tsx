import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import H1 from '../H1';

describe('<H1 />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <H1>Some big heading text.</H1>
          <H1 className="text-red-500">Some red heading red text</H1>
          <H1 className="text-[120px]">Some heading with font size 120px</H1>
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

    const treeElements = tree as ReactTestRendererJSON[];

    expect(treeElements[0].children?.length).toBe(1);
    expect(treeElements[0].children?.[0]).toBe('Some big heading text.');
    expect(treeElements[0].type).toBe('h1');
    expect(treeElements[0].props.className).toContain('text-3xl');
    expect(treeElements[0].props.className).toContain('font-medium');
    expect(treeElements[1].props.className).toContain('text-red-500');
    expect(treeElements[2].props.className).toContain('text-[120px]');
  });
});
