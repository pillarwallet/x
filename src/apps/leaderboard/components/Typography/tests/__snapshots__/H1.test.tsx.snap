// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> renders correctly 1`] = `
[
  <h1
    className="desktop:text-3xl desktop:leading-[45px] tablet:text-3xl tablet:leading-[45px] mobile:text-xl mobile:leading-5 font-medium undefined"
  >
    Some big heading text.
  </h1>,
  <h1
    className="desktop:text-3xl desktop:leading-[45px] tablet:text-3xl tablet:leading-[45px] mobile:text-xl mobile:leading-5 font-medium text-red-500"
  >
    Some red heading red text
  </h1>,
  <h1
    className="desktop:text-3xl desktop:leading-[45px] tablet:text-3xl tablet:leading-[45px] mobile:text-xl mobile:leading-5 font-medium text-[120px]"
  >
    Some heading with font size 120px
  </h1>,
]
`;
